src/collections/Table/TableRow.js

```diff
@@ -76,6 +93,20 @@ TableRow.propTypes = {
   /** A row may call attention to an error or a negative value. */
   error: PropTypes.bool,
 
+  /** An element type to render as (string or function). */
+  itemAs: customPropTypes.as,
```

Comment:
The `Label` had `detail` and `detailAs` which we merged into one:
```js
<Label detail={{ content: 'x', as: 'div' }} />
```

This is a little more tricky since `items` is an array of shorthands so it wouldn't be straightforward to add an "as" prop further up the chain. For reference, this it where it would have to be done:
```
{headerRow && <TableHeader>{TableRow.create(headerRow, { itemAs: 'th' })}</TableHeader>}
```
In this case, `headerRow` is a shorthand for a `TableRow`.